File: Data Structures & Algorithms/maximum-subarray/submission-0.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSub = nums[0] ;
        int cur = 0;
        for(int num : nums){
            if(cur < 0){
                cur = 0;
            }
            cur += num;
            maxSub = Math.max(maxSub,cur);

        }
        return maxSub;
       
        
    
        
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        result = nums[0]
        curMax,curMin = 1,1

        for num in nums:
            tmp = curMax*num
            tmp2 = curMin*num
            curMax = max(tmp,tmp2,num)
            curMin = min(tmp,tmp2,num)
            result = max(result,curMax)
        return result
        